pkg/model/components/etcd.go

```diff
@@ -27,80 +27,87 @@ const DefaultBackupImage = "kopeio/etcd-backup:1.0.20180220"
 
 // EtcdOptionsBuilder adds options for etcd to the model
 type EtcdOptionsBuilder struct {
-	Context *OptionsContext
+	*OptionsContext
 }
 
 var _ loader.OptionsBuilder = &EtcdOptionsBuilder{}
 
-const DefaultEtcdVersion = "2.2.1"
+const DefaultEtcd2Version = "2.2.1"
 
 // BuildOptions is responsible for filling in the defaults for the etcd cluster model
 func (b *EtcdOptionsBuilder) BuildOptions(o interface{}) error {
 	spec := o.(*kops.ClusterSpec)
 
 	for _, c := range spec.EtcdClusters {
+		if c.Provider == "" {
+			if b.IsKubernetesGTE("1.12") {
+				c.Provider = kops.EtcdProviderTypeManager
+			} else if c.Manager != nil {
+				c.Provider = kops.EtcdProviderTypeManager
+			} else {
+				c.Provider = kops.EtcdProviderTypeLegacy
+			}
+		}
+
 		// Ensure the version is set
-		// @TODO once we have a way of detecting a 'new' cluster we can default all clusters to v3
-		if c.Version == "" {
-			c.Version = DefaultEtcdVersion
+		if c.Version == "" && c.Provider == kops.EtcdProviderTypeLegacy {
+			// Even if in legacy mode, etcd version 2 is unsupported as of k8s 1.13
+			if b.IsKubernetesGTE("1.13") {
+				c.Version = "3.2.24"
```

Comment:
Should we make this another const to follow the same pattern as etcd2

Reply:
Good idea - done!